packages/clerk-js/src/ui/utils/createCustomMenuItems.tsx

```diff
@@ -0,0 +1,241 @@
+  return createCustomMenuItems({ customMenuItems, getDefaultMenuItems: getUserButtonDefaultMenuItems }, clerk);
+};
+
+const createCustomMenuItems = (
+  { customMenuItems, getDefaultMenuItems }: CreateCustomMenuItemsParams,
+  clerk: LoadedClerk,
+) => {
+  const { INITIAL_MENU_ITEMS, validReorderItemLabels } = getDefaultMenuItems();
+
+  const validCustomMenuItems = customMenuItems.filter(item => {
+    if (!isValidCustomMenuItem(item, validReorderItemLabels)) {
+      if (isDevelopmentSDK(clerk)) {
+        console.error('Clerk: Invalid custom menu item:', item);
+      }
+      return false;
+    }
+    return true;
+  });
+
+  if (isDevelopmentSDK(clerk)) {
+    checkForDuplicateUsageOfReorderingItems(customMenuItems, validReorderItemLabels);
+    warnForDuplicateLabels(validCustomMenuItems);
+  }
+
+  const { menuItems } = getMenuItems({ customMenuItems: validCustomMenuItems, defaultMenuItems: INITIAL_MENU_ITEMS });
+
+  return menuItems;
+};
+
+const getMenuItems = ({
+  customMenuItems,
+  defaultMenuItems,
+}: {
+  customMenuItems: CustomMenuItem[];
+  defaultMenuItems: MenuItem[];
+}) => {
+  let remainingDefaultItems: MenuItem[] = defaultMenuItems.map(r => r);
+
+  const items: MenuItem[] = customMenuItems.map((ci: CustomMenuItem, index) => {
+    if (isCustomMenuLink(ci)) {
+      return {
+        name: ci.label,
+        id: `custom-menutItem-${index}`,
+        icon: props => (
+          <ExternalElementMounter
+            mount={ci.mountIcon}
+            unmount={ci.unmountIcon}
+            {...props}
+          />
+        ),
+        path: sanitizeCustomLinkURL(ci.href),
+        external: true,
+      };
+    }
+
+    if (isCustomMenuItem(ci)) {
+      return {
+        name: ci.label,
+        id: `custom-menutItem-${index}`,
+        icon: props => (
+          <ExternalElementMounter
+            mount={ci.mountIcon}
+            unmount={ci.unmountIcon}
+            {...props}
+          />
+        ),
+        onClick: ci?.onClick,
+      };
+    }
+
+    if (isCustomMenuItemWithOpen(ci)) {
+      return {
+        name: ci.label,
+        id: `custom-menutItem-${index}`,
+        icon: props => (
+          <ExternalElementMounter
+            mount={ci.mountIcon}
+            unmount={ci.unmountIcon}
+            {...props}
+          />
+        ),
+        open: ci.open,
+      };
+    }
+
+    const reorderItem = defaultMenuItems.find(item => item.id === ci.label) as MenuItem;
+    remainingDefaultItems = remainingDefaultItems.filter(item => item.id !== ci.label);
+
+    return { ...reorderItem };
+  });
+
+  const allItems = [...items, ...remainingDefaultItems];
+
+  const hasReorderedManageAccountOrCreateOrg = customMenuItems.some(
+    item => item.label === USER_BUTTON_ITEM_ID.MANAGE_ACCOUNT,
+  );
+  // Ensure that the "Manage account" and "Create organization" item is always at the top of the list if it's not included in the custom items
```

Comment:
❓Does this affect orgs ?
```suggestion
  // Ensure that the "Manage account" item is always at the top of the list if it's not included in the custom items
```

Reply:
Done